app/concerns/persistable.rb

```diff
@@ -1,6 +1,15 @@
 module Persistable
 
   module ClassMethods
+    def attributes
+      self::ATTRIBUTES
+    end
+
+    def self.extended(base)
+      base.attributes.keys.each do |attribute_name|
+        base.send(:attr_accessor, attribute_name)
+      end
+    end
 
```

Comment:
When I mixin this class module, I call the `Class.attributes` reader I defined above to expose the `ATTRIBUTES` hash and then I use `send` to call `attr_accessor` on the class and define the attributes.